Non-fatal error when ET_SWITCH encounters unsupported dtype

manuelcandales opened the pull request:
Summary: Modify `ET_INTERNAL_SWITCH` macro, so that we fail non-fatally in the default clause when we encounter an unsupported dtype.

Differential Revision: D80141272

digantdesai commented on runtime/core/exec_aten/util/scalar_type_util.h:
```diff
@@ -910,20 +910,21 @@ struct promote_types {
+    switch (_st) {                                             \
+      __VA_ARGS__                                              \
+      default:                                                 \
+        CONTEXT.fail(torch::executor::Error::InvalidArgument); \
```
Nit: Leave a comment along the lines "this sets the return code which should be checked by the caller"

manuelcandales committed:
Non-fatal error when ET_SWITCH encounters unsupported dtype (#13359)

Summary:

Modify `ET_INTERNAL_SWITCH` macro, so that we fail non-fatally in the default clause when we encounter an unsupported dtype.

Reviewed By: digantdesai

Differential Revision: D80141272
